docs/tutorial-run-solver.md

```diff
@@ -0,0 +1,176 @@
+5. [Run the Solver](#run-the-solver)
+6. [Advanced Configuration](#advanced-configuration)
+   1. [Orderbook Filter](#Configuring-an-Orderbook-Filter)
+
+---
+
+## Installing Driver Tools
+
+Clone the dex-services repository, and acquire the project requirements
+
+```sh
+git clone git@github.com:gnosis/dex-services.git
+```
+
+We demonstrate how to run the open-solver. Once comfortable, the open-solver can be adapted to suit your own needs from the open-solver repo
+
+```sh
+git clone git@github.com:gnosis/dex-solver.git
+```
+
+## Building The Driver
+
+Build the driver (with open-solver dockerhub image) from within the root directory of `dex-services` repo. This command builds the driver while simultanesouly baking the open solver into the the build.
+
+```sh
+docker-compose build --build-arg SOLVER_BASE=gnosispm/dex-open-solver:master stablex-debug
+```
+
+Note that, if `SOLVER_BASE` is not specified, the driver's own internal `naive-solver` will be used. Alternatively, if you have your own personalized solver, this image would need to be specified during the build.
+We will see that specifying solver type again at runtime is also expected and this may seem counterintuitive.
+However, the argument used at runtime is meant to specify potentially different implementations from within the existing solver build.
+For example, you may have a `standard-solver` which matches orders according to one algorithm and another, say `fallback-solver` which makes use of a fixed list of currated tokens or simply operates in a different way.
+
+## Configuring Environment Variables
+
+The following three environment variables are required when running the dex-services project.
+Note that the account corresponding to the private key should be funded with sufficient ETH to pay gas for solution submission.
+
+```sh
+export PRIVATE_KEY=<YOUR_PRIVATE_KEY>
+export ETHEREUM_NODE_URL=https://mainnet.infura.io/v3/<YOUR_INFURA_KEY>
+export NETWORK_ID=<YOUR_PREFERED_NETWORK_ID>
+```
+
+By placing these values into an `.env` file and sourcing this file, you can avoid adding them to as runtime arguments on every execution.
+
+## Acquiring the Orderbook
+
+It can take a very long time for the driver to load the orderbook and this process is also limited by infura rate limits so it is best to supply an existing orderbook file, available for download at
+
+- [mainnet](https://gnosis-dfusion-volume-mainnet.s3.amazonaws.com/stablex_orderbook_mainnet.bin)
+- [rinkeby](https://gnosis-dfusion-volume-rinkeby.s3.amazonaws.com/stablex_orderbook_rinkeby.bin)
+
+Copy this orderbook file into `dex-services/target` and point your solver to this file via the runtime argument `--orderbook-file` or environment vairable `ORDERBOOK_FILE`. That is, you can append `ORDERBOOK_FILE=/app/dex-services/target/stablex_orderbook_mainnet.bin` to your `.env` file.
+
+## Run the Solver
+
+First run and enter the solver container
+
+```sh
+docker-compose run -v $(PWD)/:/app/dex-services stablex-debug
```

Comment:
All env variables should be set inside the Docker image. From the tutorial I thought it was enough to set them in the base system. After this command it's good to source `.env`. (Which should be in the `dex-services` folder.)

Error I got otherwise:
```
2020-09-09T14:01:06.980Z ERRO [services_core::logging] thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: web3 error: Transport error: CouldntResolveHost: couldn't resolve host name
```

Reply:
Right, the env file should be sourced from within the container. I will try and make this more clear.